wagtail/admin/templates/wagtailadmin/shared/button.stories.tsx

```diff
@@ -0,0 +1,218 @@
+      <svg className="icon icon-cog icon" aria-hidden="true">
+        <use href="#icon-cog" />
+      </svg>
+      button link
+    </a>
+    <button
+      type="button"
+      className="button button-small text-replace button--icon"
+    >
+      <svg className="icon icon-cog icon" aria-hidden="true">
+        <use href="#icon-cog" />
+      </svg>
+      button element
+    </button>
+
+    <h3>Colour signifiers</h3>
+
+    <h4>Positive</h4>
+
+    <a href="#" className="button yes">
+      yes
+    </a>
+    <a href="#" className="button button-small yes">
+      yes
+    </a>
+
+    <h4>Negative</h4>
+
+    <a href="#" className="button no">
+      No
+    </a>
+    <a href="#" className="button button-small no">
+      No
+    </a>
+
+    <h3>
+      Buttons with internal loading indicators (currently only{' '}
+      <code>button</code> supported)
+    </h3>
+    <button type="button" className="button button-longrunning">
+      <svg className="icon icon-spinner icon" aria-hidden="true">
+        <use href="#icon-spinner" />
+      </svg>
+      Click me
+    </button>
+
+    <h4>Secondary</h4>
+    <button
+      type="button"
+      className="button button-secondary button-longrunning"
+    >
+      <svg className="icon icon-spinner icon" aria-hidden="true">
+        <use href="#icon-spinner" />
+      </svg>
+      Click me
+    </button>
+
+    <h4>Small</h4>
+    <button type="button" className="button button-small button-longrunning">
+      <svg className="icon icon-spinner icon" aria-hidden="true">
+        <use href="#icon-spinner" />
+      </svg>
+      Click me
+    </button>
+
+    <h4>Buttons where the text is replaced on click</h4>
+    <button
+      type="button"
+      className="button button-longrunning"
+      data-clicked-text="Test"
+    >
+      <svg className="icon icon-spinner icon" aria-hidden="true">
+        <use href="#icon-spinner" />
+      </svg>
+      <em>Click me</em>
+    </button>
+
+    <h3>Mixtures</h3>
+
+    <button type="button" className="button button--icon text-replace yes">
+      <svg className="icon icon-tick icon" aria-hidden="true">
+        <use href="#icon-tick" />
+      </svg>
+      A proper button
+    </button>
+    <a href="#" className="button button--icon text-replace white">
+      <svg className="icon icon-cog icon" aria-hidden="true">
+        <use href="#icon-cog" />
+      </svg>
+      A link button
+    </a>
+    <span className="button button--icon text-replace no ">
+      <svg className="icon icon-bin icon" aria-hidden="true">
+        <use href="#icon-bin" />
+      </svg>
+      A non-link button
+    </span>
+    <a href="#" className="button button--icon bicolor disabled">
+      <span className="icon-wrapper">
+        <svg className="icon icon-tick icon" aria-hidden="true">
+          <use href="#icon-tick" />
+        </svg>
+      </span>
+      button link
+    </a>
+  </section>
+);
```

Comment:
```

export const All = Template.bind({});

All.args = { url: window.location.toString() };

```

export down here instead with the `All.args` approach - and give us a URL to use for links.